### DISCUSSION TITLE
32bits platform

### DISCUSSION THREAD

**Discussion by Agg242** [stclib/STC#77] [General]
Hi,
As my Amiga with a 680x0 cpu works with 32bits ints at best, I am wondering if the 64 bits types for ranges, hashes and random number generator is not penalizing.
I've put conditional defines to detect compilation with the Amiga and lower the _llong type, but I guess messing with the hash functions will not be good.
I noticed that the crange_value type is defined as long long, just after including common.h which defines _llong for use with c_forrange_x values. Wouldn't it be interesting to use the available _llong type  ?

**Comment by tylov** [comment]
Although this issue is a year old, in the upcoming V5.0 (release December) both ranges and hashes will be 32-bit on 32-bit hardware. You can also explicitly use 32-bit ranges on 64-bit hardware for increased performance.

**Reply by Agg242** [reply]
This is excellent news !
Can't wait for the release, as I'll test it with AdventOfCode challenges with my Amiga500.

**Reply by tylov** [reply]
Haha, was just doing some AoC with STC myself ! Good fun at this time of year.
